tools/udisksctl.c

```diff
@@ -575,6 +575,19 @@ static const GOptionEntry command_unmount_entries[] =
   }
 };
 
+static void
+list_options (const GOptionEntry *entries)
+{
+  for (const GOptionEntry* current = entries; current->long_name; current++)
+    {
+      if ((current->arg != G_OPTION_ARG_STRING && current->arg != G_OPTION_ARG_FILENAME) ||
+          current->arg_data != NULL)
```

Comment:
Yes, you're right. I remember thinking about it but struggling to find a test scenario to check it at the time. For both STRING and FILENAME arguments, it is a `gchar*` via `gpointer`. But I can't persuade it to work with a comparison against the NULL macro rather than the traditional C approach of using the pointer as a boolean directly. I think the intent of the code is to avoid completing options that have already been specified. It can be exercised with:

```
udisksctl complete 'u mount --object-path x x ' 30
```
I'm guessing the intent of the check is for that to list all options except `--object-path` because that's already there so no need to complete it. That is now working but if you're unhappy with the form of the condition, I'll need some hints on getting it to work because it doesn't with `NULL` and `(void *)0` generates a warning.

Similar handling might be done for other types if you make assumptions about the initial default values. Repeatable options could also be problematic – if g_option even supports them.

I don't know why the second x (and 30 instead of 28) is needed in that test case. In general, the bash completion doesn't appear to work beyond the first argument but that is beyond the scope of this pull request because it didn't work before either.

Reply:
Took me a while to decipher it but I think it's correct.

> I'll need some hints on getting it to work because it doesn't with NULL and (void *)0 generates a warning.
I think it needs to properly enclose the test in parenthesis for the compiler. But the current form looks good.